src/java.base/share/classes/java/net/URLConnection.java

```diff
@@ -1180,6 +1192,16 @@ public String getRequestProperty(String key) {
      * of Strings that represents the corresponding
      * field values.
      *
+     * If multiple values for a given key are added via the
+     * {@link #addRequestProperty(String, String)} method,
+     * these values will be returned in the order they were
+     * added. This method must preserve the insertion order
+     * of such values.
+     *
```

Comment:
But we have made addRequestProperty effectively a no-op ( as above ) !

Reply:
We are allowing it to be a no-op. But if it's not a no-op then it must return values in the order they were added. The other option is to make the order non normative and protocol dependent (and possibly move the normative text to the HttpURLConnection/HttpsURLConnection/JarURLConnection.